### PR TITLE
Odf slicer

### PR DESCRIPTION
**Pull request by Garyfallidis**

### CODE REVIEW

File: dipy/viz/actor.py [hunk 1]
```diff
@@ -539,3 +546,206 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
...
+        Distance between spheres.
+    norm : bool
+        Normalize `sphere_values`.
+    radial_scale : bool
+        Scale sphere points according to odf values.
+    opacity : float
+        Takes values from 0 (fully transparent) to 1 (opaque)
+    colormap : None or str
+        If None then white color is used. Otherwise the name of colormap is
+        given. Matplotlib colormaps are also supported. Try 'inferno' :).
```
Comment (arokem) [inline dipy/viz/actor.py]: How about => "Matplotlib colormaps are supported (e.g., 'inferno')"

File: dipy/viz/actor.py [hunk 2]
```diff
@@ -539,3 +546,206 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
...
+        If True the colormap will be applied in all ODFs. If False
+        it will be applied individually at each voxel (default False).
+    """
+
+    if mask is None:
+        mask = np.ones(odfs.shape[:3], dtype=np.bool)
+    else:
+        mask = mask.astype(np.bool)
+
+    I, J, K = odfs.shape[:3]
```
Comment (arokem) [inline dipy/viz/actor.py]: PEP8: lower-case for these.

File: dipy/viz/actor.py [hunk 3]
```diff
@@ -539,3 +546,206 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
...
+        Distance between spheres.
+    norm : bool
+        Normalize `sphere_values`.
+    radial_scale : bool
+        Scale sphere points according to odf values.
+    opacity : float
+        Takes values from 0 (fully transparent) to 1 (opaque)
+    colormap : None or str
+        If None then white color is used. Otherwise the name of colormap is
+        given. Matplotlib colormaps are also supported. Try 'inferno' :).
```
Comment (arokem) [inline dipy/viz/actor.py]: Same comment as above.

File: dipy/viz/actor.py [hunk 4]
```diff
@@ -539,3 +546,325 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
...
+
+    all_xyz = []
+    all_faces = []
+    all_ms = []
+    for (k, center) in enumerate(ijk):
+
+        m = odfs[tuple(center.astype(np.int))].copy()
+
+        if norm:
+            m /= abs(m).max()
```
Comment (arokem) [inline dipy/viz/actor.py]: Should this be `np.abs`?

File: dipy/viz/actor.py [hunk 5]
```diff
@@ -539,3 +546,325 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
...
+        if radial_scale:
+            xyz = vertices * m[:, None]
+        else:
+            xyz = vertices.copy()
+
+        all_xyz.append(scale * xyz + center)
+        all_faces.append(faces + k * xyz.shape[0])
+        all_ms.append(m)
+
+    # from ipdb import set_trace
```
Comment (arokem) [inline dipy/viz/actor.py]: Please remove debugging statement

File: dipy/viz/actor.py [hunk 6]
```diff
@@ -539,3 +546,325 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
...
+    # set_trace()
```
Comment (arokem) [inline dipy/viz/actor.py]: And here.

File: dipy/viz/actor.py [hunk 7]
```diff
@@ -539,3 +546,325 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
...
+                            dtype='f4')
+            for k in range(ijk.shape[0]):
+                tmp = create_colormap(all_ms[k].ravel(), colormap)
+                cols[k] = tmp.copy()
+
+            cols = np.ascontiguousarray(
+                np.reshape(cols, (cols.shape[0] * cols.shape[1],
+                           cols.shape[2])), dtype='f4')
+        # cols = np.interp(cols, [0, 1], [0, 255]).astype('ubyte')
+        # vtk_colors = numpy_to_vtk_colors(255 * cols)
```
Comment (arokem) [inline dipy/viz/actor.py]: Please remove these commented lines

File: dipy/viz/actor.py [hunk 8]
```diff
@@ -539,3 +546,325 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
...
+            if x is not None:
+                self.display_extent(x, x, 0, J - 1, 0, K - 1)
+            if y is not None:
+                self.display_extent(0, I - 1, y, y, 0, K - 1)
+            if z is not None:
+                self.display_extent(0, I - 1, 0, J - 1, z, z)
+
+    peak_actor = PeakSlicerActor()
+
+    I, J, K = grid_shape
```
Comment (arokem) [inline dipy/viz/actor.py]: PEP8: lower-case

File: dipy/viz/colormap.py [hunk 9]
```diff
@@ -255,3 +262,58 @@ def line_colors(streamlines, cmap='rgb_standard'):
                     for streamline in streamlines]
 
     return np.vstack(col_list)
+
+
+lowercase_cm_name = {'blues': 'Blues', 'accent': 'Accent'}
+
+
+def create_colormap(v, name='jet', auto=True):
```
Comment (arokem) [inline dipy/viz/colormap.py]: Can we set the default to `plasma`?

**Comment by Garyfallidis** [comment]
@arokem I made all the changes that you requested. @skoudoro or @arokem please merge this PR when travis finishes successfully. Ignore coverage for this PR. Go go team!

**Comment by arokem** [comment]
I halted all but the last Travis build, in hopes that this will speed things up.

File: dipy/viz/actor.py (hunk 1 above)
Comment (arokem) [inline dipy/viz/actor.py]: This still needs to be changed.

**Comment by Garyfallidis** [comment]
Cool, thanks! Good luck with the seminar at OHBM.

**Pull request merged**